### ISSUE TITLE
Keybase needs distro package support

### ISSUE THREAD

**Issue by danielquinn** [keybase/keybase-issues#377]
While npm is nifty, expecting someone to install node (typically as a package) and then su to root to run an arbitrary command that must then be maintained outside of their distro's package management system is a little over the top.

At the very least, it would be good to support Debian/Ubuntu, Redhat/SuSe, and Gentoo so you have most of the bases covered.

**Edit**: As suggested, I've added a checklist of the various packagers that could be supported:
- [ ] Ubuntu / Debian
- [ ] Redhat / Suse
- [x] Gentoo (ebuild on this ticket)
- [x] [Arch](https://aur.archlinux.org/packages/keybase/)
- [ ] Windows

**Comment by jaysonsantos** [comment]
I started a simple gentoo ebuild [1], I only can commit in sunrise overlay or I can ask a developer to put me as proxy-mantainer.
[1] https://github.com/jaysonsantos/overlay/blob/master/app-crypt/keybase/keybase-0.7.7.ebuild

**Comment by danielquinn** [comment]
Excellent work @jaysonsantos!  Honestly, this is all that is really required (at least for me) to make keybase usable on a Linux system.  And on my favourite distro too!

If we ever meet in person, I shall buy you a beer kind sir.

**Comment by dtiersch** [comment]
I don't think that any packaging work should be invested into the client right now, because it is being rewritten in Go and with that node and npm become obsolete.
When the Go client is ready I'm going to set some debian packaging stuff up and try to make it semi-automatic (except for the keysigning part - of course), but right now I consider it wasted effort...

**Comment by maxtaco** [comment]
Hey guys! Thanks for the work here and this great discussion.  We're working as hard as possible on the new Go client but we're rolling out a ton of new features so it's not going to be ready right away.  Closer every day.  I still plan on supporting the node client until we have the new client ready.  So I don't think it's wasted effort to integrate with package managers.  The more users the better, for sure, it will make the service better for everyone.

**Comment by jaysonsantos** [comment]
@danielquinn It is not in sunrise yet, I will submit a bug report today after some reviews.
@dtiersch In gentoo side that does not matter (if the go version is incremented after the node version) because every ebuild version is a different file.

**Comment by jaysonsantos** [comment]
I am concerned about the node-client license, does anyone know what it is?

**Comment by maxtaco** [comment]
BSD-3-Clause (new BSD)

**Comment by frank-borkin** [comment]
Hi all,
I've created a PPA (actually, a bit more than a week ago and forgot to tell anyone): 
sudo add-apt-repository ppa:borkinfr/keybase
sudo apt-get update
sudo apt-get install keybase

It's a bit fugly right now, but it works (at least on trusty). I'll get some improvements in over the next few weeks and cover the other versions (utopic, etc). I don't mind changing it to a Go client when one comes out, but I'm very interested to how you do versioning and if there's a numbered changelog anywhere?

Frank

**Comment by dtiersch** [comment]
The Changelog is [here](https://github.com/keybase/node-client/blob/master/CHANGELOG.md).

**Comment by n4cer** [comment]
fyi the nodeclient is in the gentoo portage tree now.

**Comment by maxtaco** [comment]
Thanks!

**Comment by zekesonxx** [comment]
Someone made [an AUR package](https://aur.archlinux.org/packages/keybase/) for keybase two days after this issue was posted, so you can check Arch off.

**Comment by aspiers** [comment]
I just submitted #2569 moments before discovering this issue.  As per #2569, please let me know how I can help with packaging the new keybase client for openSUSE!  (And please consider the possibility of using the [Open Build Service](http://openbuildservice.org/) to build packages for [many distributions](https://en.opensuse.org/openSUSE:Build_Service_supported_build_targets) from a single source package ...)

**Comment by Saklad5** [comment]
Could you provide a versioned download source for the Keybase GUI client so it works properly with Homebrew-Cask?
